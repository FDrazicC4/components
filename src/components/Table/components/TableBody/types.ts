import { ReactNode } from 'react';


export interface Props {


    /**
     * Any custom content to fill the TableBody that is different than default
     */
    children?: ReactNode,


    /**
     * The className for the TableBody
     */
    className?: string,


    /**
     * The row data used to populate the TableBody
     */
    rows?: ReactNode[],


}
