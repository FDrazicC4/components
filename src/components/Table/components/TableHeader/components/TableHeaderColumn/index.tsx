import React, { ReactElement } from 'react';
import clsx from 'clsx';
import { Props } from './types';


/**
 * The `<TableHeaderColumn />` component represents a single column in the `<TableHeader />`.
 */
function TableHeaderColumn({
    children,
    className,
    label,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableHeaderColumn',
        className,
    );


    const content = children || label;


    return (
        <th className={classes} {...otherProps}>
            {content}
        </th>
    );


}


export default TableHeaderColumn;
