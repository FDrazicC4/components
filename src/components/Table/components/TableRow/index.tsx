import React, { ReactElement } from 'react';
import clsx from 'clsx';
import { Props } from './types';


/**
 * The `<TableRow />` component
 */
function TableRow({
    className,
    children,
    cells,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableRow',
        className,
    );


    const data = children || (cells?.map((cell: ReactElement, index: number) => (
        <td key={index}>
            {cell}
        </td>
    )));


    return (
        <tr className={classes} {...otherProps}>
            {data}
        </tr>
    );


}


export default TableRow;
