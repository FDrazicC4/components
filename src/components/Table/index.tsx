import React from 'react';
import clsx from 'clsx';
import { Props } from './types';
import MuiTable from '@material-ui/core/Table';


const Table = React.forwardRef(({ className, ...props }: Props, ref: React.Ref<any>): React.ReactElement => {


    const classes = clsx(
        'brew-Table',
        className,
    );

    return (
        <MuiTable className={classes} ref={ref} {...props} />
    );

});


Table.displayName = 'BrewkitTable';

export default Table;
