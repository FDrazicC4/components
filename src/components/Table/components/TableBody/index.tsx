import React, { ReactElement } from 'react';
import clsx from 'clsx';
import { Props } from './types';
import TableRow from '../TableRow';


/**
 * The `<TableBody />` component represents the standard body of a `<Table />` component.
 */
function TableBody({
    className,
    children,
    rows,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableBody',
        className,
    );


    const content = children || (rows?.map((row: ReactElement[], index: number) => (
        <TableRow cells={row} key={index} />
    )));


    return (
        <tbody className={classes} {...otherProps}>
            {content}
        </tbody>
    );


}


export default TableBody;
