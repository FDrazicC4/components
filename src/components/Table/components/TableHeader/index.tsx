import React, { ReactElement } from 'react';
import clsx from 'clsx';
import TableHeaderColumn from './components/TableHeaderColumn';
import { Props as ColumnProps } from './components/TableHeaderColumn/types';
import TableRow from '../TableRow';
import { Props } from './types';


/**
 * The `<TableHeader />` component represents the standard header of a `<Table />` component.
 */
function TableHeader({
    children,
    className,
    columns,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableHeader',
        className,
    );


    const content = children || (columns?.map((colData: ColumnProps, index: number) => (
        <TableHeaderColumn key={index} {...colData} />
    )));


    return (
        <thead className={classes} {...otherProps}>
            <TableRow>
                {content}
            </TableRow>
        </thead>
    );


}

TableHeader.Column = TableHeaderColumn;

export default TableHeader;
