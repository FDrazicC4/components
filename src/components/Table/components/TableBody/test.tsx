import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Table from '../../index';
import TableBody from './index';
import TableRow from '../TableRow';
import Typography from '../../../Typography';


describe('Table Body', () => {


    it('Renders correctly', () => {
        const content = renderer
            .create((
                <TableBody>
                    <TableRow>
                        <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>
                    </TableRow>
                    <TableRow>
                        <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                    </TableRow>
                </TableBody>
            ))
            .toJSON();

        expect(content).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <Table>
                <TableBody className={className}>
                    <TableRow>
                        <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>

                    </TableRow>
                    <TableRow>
                        <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                    </TableRow>
                </TableBody>
            </Table>
        ));

        expect(content.children().first()
            .shallow()
            .hasClass(className))
            .toBe(true);
        expect(content.children().first()
            .shallow()
            .hasClass('brew-TableBody'))
            .toBe(true);
    });


});


describe('Table Body Props', () => {


    it('Renders "rows" prop correctly', () => {
        const content = renderer
            .create((
                <TableBody
                    rows={[
                        ['row1-col1', 'row1-col2', 'row1-col3'],
                        ['row2-col1', 'row2-col2', 'row2-col3'],
                        ['row3-col1', 'row3-col2', 'row3-col3'],
                    ]}
                />
            ))
            .toJSON();

        expect(content).toMatchSnapshot();

        expect(mount((
            <Table>
                <TableBody
                    rows={[
                        [
                            <Typography key="row1-col1" variant="body2">Row1-Col1</Typography>,
                            <Typography key="row1-col2" variant="body2">Row1-Col2</Typography>,
                            <Typography key="row1-col3" variant="body2">Row1-Col3</Typography>,
                        ],
                        [
                            <Typography key="row2-col1" variant="body2">Row2-Col1</Typography>,
                            <Typography key="row2-col2" variant="body2">Row2-Col2</Typography>,
                            <Typography key="row2-col3" variant="body2">Row2-Col3</Typography>,
                        ],
                        [
                            <Typography key="row3-col1" variant="body2">Row3-Col1</Typography>,
                            <Typography key="row3-col2" variant="body2">Row3-Col2</Typography>,
                            <Typography key="row3-col3" variant="body2">Row3-Col3</Typography>,
                        ],
                    ]}
                />
            </Table>
        ))
            .find(Typography)
            .find('[variant="body2"]')
            .children()
            .contains('Row1-Col3')).toBe(true);
    });


});
