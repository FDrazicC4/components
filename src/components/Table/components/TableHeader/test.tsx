import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Table from '../../index';
import TableHeader from './index';
import TableHeaderColumn from './components/TableHeaderColumn';
import Typography from '../../../Typography';


describe('Table Header', () => {


    it('Renders correctly', () => {
        const header = renderer
            .create((
                <TableHeader>
                    <TableHeaderColumn>Column1</TableHeaderColumn>
                    <TableHeaderColumn>Column2</TableHeaderColumn>
                    <TableHeaderColumn>Column3</TableHeaderColumn>
                </TableHeader>
            )).toJSON();

        expect(header).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <Table>
                <TableHeader className={className}>
                    <TableHeaderColumn>
                        Column4
                    </TableHeaderColumn>
                    <TableHeaderColumn>
                        Column5
                    </TableHeaderColumn>
                    <TableHeaderColumn>
                        Column6
                    </TableHeaderColumn>
                </TableHeader>
            </Table>
        ));

        expect(content.children().first()
            .shallow()
            .hasClass(className))
            .toBe(true);
        expect(content.children().first()
            .shallow()
            .hasClass('brew-TableHeader'))
            .toBe(true);
    });


});

describe('Table Header Props', () => {


    it('Renders "columns" prop correctly', () => {
        const header = renderer
            .create((
                <TableHeader
                    columns={[
                        { label: 'ColumnA' },
                        { label: 'ColumnB' },
                        { label: 'ColumnC' },
                    ]}
                />
            ))
            .toJSON();

        expect(header).toMatchSnapshot();

        expect(mount((
            <Table>
                <TableHeader
                    columns={[
                        { label: <Typography variant="h3">ColumnA</Typography> },
                        { label: <Typography variant="h3">ColumnB</Typography> },
                        { label: <Typography variant="h3">ColumnC</Typography> },
                    ]}
                />
            </Table>
        ))
            .find(Typography)
            .find('[variant="h3"]')
            .children()
            .contains('ColumnB')).toBe(true);
    });

});
