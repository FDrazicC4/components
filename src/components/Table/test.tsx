import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Table from './index';
import TableBody from './components/TableBody';
import TableHeader from './components/TableHeader';
import TableHeaderColumn from './components/TableHeader/components/TableHeaderColumn';
import TableRow from './components/TableRow';
import Typography from '../Typography';


describe('Table', () => {

    it('Renders correctly', () => {
        const content = renderer
            .create((
                <Table>
                    <TableHeader>
                        <TableHeaderColumn>Column1</TableHeaderColumn>
                        <TableHeaderColumn>Column2</TableHeaderColumn>
                        <TableHeaderColumn>Column3</TableHeaderColumn>
                    </TableHeader>
                    <TableBody>
                        <TableRow>
                            <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>
                        </TableRow>
                        <TableRow>
                            <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                        </TableRow>
                    </TableBody>
                </Table>
            ))
            .toJSON();
        expect(content).toMatchSnapshot();
    });

    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <Table className={className}>
                <TableHeader>
                    <TableHeaderColumn>Column1</TableHeaderColumn>
                    <TableHeaderColumn>Column2</TableHeaderColumn>
                    <TableHeaderColumn>Column3</TableHeaderColumn>
                </TableHeader>
                <TableBody>
                    <TableRow>
                        <td>Row1-Col1</td><td>Row1-Col2</td><td>Row1-Col3</td>
                    </TableRow>
                    <TableRow>
                        <td>Row2-Col1</td><td>Row2-Col2</td><td>Row2-Col3</td>
                    </TableRow>
                </TableBody>
            </Table>
        ));

        expect(content.hasClass(className)).toBe(true);
        expect(content.hasClass('brew-Table')).toBe(true);
    });


});

// eslint-disable-next-line max-lines-per-function
describe('Table Props', () => {


    it('Renders "columns" prop correctly', () => {
        const content = renderer
            .create((
                <Table
                    columns={[
                        { label: 'ColumnA' },
                        { label: 'ColumnB' },
                        { label: 'ColumnC' },
                    ]}
                />
            ))
            .toJSON();
        expect(content).toMatchSnapshot();

        expect(mount((
            <Table
                columns={[
                    { label: <Typography variant="body1">ColumnA</Typography> },
                    { label: <Typography variant="body1">ColumnB</Typography> },
                    { label: <Typography variant="body1">ColumnC</Typography> },
                ]}
            />
        ))
            .find(Typography)
            .find('[variant="body1"]')
            .children()
            .contains('ColumnA')).toBe(true);
    });


    it('Renders "rows" prop correctly', () => {
        const content = renderer
            .create((
                <Table
                    rows={[
                        ['row1-col1', 'row1-col2', 'row1-col3'],
                        ['row2-col1', 'row2-col2', 'row2-col3'],
                        ['row3-col1', 'row3-col2', 'row3-col3'],
                    ]}
                />
            ))
            .toJSON();
        expect(content).toMatchSnapshot();

        expect(mount((
            <Table
                rows={[
                    [
                        <Typography key="row1-col1" variant="body2">Row1-Col1</Typography>,
                        <Typography key="row1-col2" variant="body2">Row1-Col2</Typography>,
                        <Typography key="row1-col3" variant="body2">Row1-Col3</Typography>,
                    ],
                    [
                        <Typography key="row2-col1" variant="body2">Row2-Col1</Typography>,
                        <Typography key="row2-col2" variant="body2">Row2-Col2</Typography>,
                        <Typography key="row2-col3" variant="body2">Row2-Col3</Typography>,
                    ],
                    [
                        <Typography key="row3-col1" variant="body2">Row3-Col1</Typography>,
                        <Typography key="row3-col2" variant="body2">Row3-Col2</Typography>,
                        <Typography key="row3-col3" variant="body2">Row3-Col3</Typography>,
                    ],
                ]}
            />
        ))
            .find(Typography)
            .find('[variant="body2"]')
            .children()
            .contains('Row3-Col3')).toBe(true);
    });

    it('Renders "columns" and "rows" prop correctly simultaneously', () => {
        const content = renderer
            .create((
                <Table
                    columns={[
                        { label: 'ColumnX' },
                        { label: 'ColumnY' },
                        { label: 'ColumnZ' },
                    ]}
                    rows={[
                        ['row1-colX', 'row1-colX', 'row1-colX'],
                        ['row2-colY', 'row2-colY', 'row2-colY'],
                        ['row3-colZ', 'row3-colZ', 'row3-colZ'],
                    ]}
                />
            ))
            .toJSON();
        expect(content).toMatchSnapshot();

        expect(mount((
            <Table
                columns={[
                    { label: <Typography variant="h3">ColumnX</Typography> },
                    { label: <Typography variant="h3">ColumnY</Typography> },
                    { label: <Typography variant="h3">ColumnZ</Typography> },
                ]}
                rows={[
                    [
                        <Typography key="row1-colx" variant="body2">Row1-ColX</Typography>,
                        <Typography key="row1-coly" variant="body2">Row1-ColY</Typography>,
                        <Typography key="row1-colz" variant="body2">Row1-ColZ</Typography>,
                    ],
                    [
                        <Typography key="row2-colx" variant="body2">Row2-ColX</Typography>,
                        <Typography key="row2-coly" variant="body2">Row2-ColY</Typography>,
                        <Typography key="row2-colz" variant="body2">Row2-ColZ</Typography>,
                    ],
                    [
                        <Typography key="row3-colx" variant="body2">Row3-ColX</Typography>,
                        <Typography key="row3-coly" variant="body2">Row3-ColY</Typography>,
                        <Typography key="row3-colz" variant="body2">Row3-ColZ</Typography>,
                    ],
                ]}
            />
        ))
            .find(Typography)
            .find('[variant="body2"]')
            .children()
            .contains('Row2-ColY')).toBe(true);
    });


});
