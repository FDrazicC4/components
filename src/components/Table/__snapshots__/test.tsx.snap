// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Props Renders "columns" and "rows" prop correctly simultaneously 1`] = `
<table
  className="brew-Table"
>
  <thead
    className="brew-TableHeader"
  >
    <tr
      className="brew-TableRow"
    >
      <th
        className="brew-TableHeaderColumn"
      >
        ColumnX
      </th>
      <th
        className="brew-TableHeaderColumn"
      >
        ColumnY
      </th>
      <th
        className="brew-TableHeaderColumn"
      >
        ColumnZ
      </th>
    </tr>
  </thead>
  <tbody
    className="brew-TableBody"
  >
    <tr
      className="brew-TableRow"
    >
      <td>
        row1-colX
      </td>
      <td>
        row1-colX
      </td>
      <td>
        row1-colX
      </td>
    </tr>
    <tr
      className="brew-TableRow"
    >
      <td>
        row2-colY
      </td>
      <td>
        row2-colY
      </td>
      <td>
        row2-colY
      </td>
    </tr>
    <tr
      className="brew-TableRow"
    >
      <td>
        row3-colZ
      </td>
      <td>
        row3-colZ
      </td>
      <td>
        row3-colZ
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Props Renders "columns" prop correctly 1`] = `
<table
  className="brew-Table"
>
  <thead
    className="brew-TableHeader"
  >
    <tr
      className="brew-TableRow"
    >
      <th
        className="brew-TableHeaderColumn"
      >
        ColumnA
      </th>
      <th
        className="brew-TableHeaderColumn"
      >
        ColumnB
      </th>
      <th
        className="brew-TableHeaderColumn"
      >
        ColumnC
      </th>
    </tr>
  </thead>
  <tbody
    className="brew-TableBody"
  />
</table>
`;

exports[`Table Props Renders "rows" prop correctly 1`] = `
<table
  className="brew-Table"
>
  <thead
    className="brew-TableHeader"
  >
    <tr
      className="brew-TableRow"
    />
  </thead>
  <tbody
    className="brew-TableBody"
  >
    <tr
      className="brew-TableRow"
    >
      <td>
        row1-col1
      </td>
      <td>
        row1-col2
      </td>
      <td>
        row1-col3
      </td>
    </tr>
    <tr
      className="brew-TableRow"
    >
      <td>
        row2-col1
      </td>
      <td>
        row2-col2
      </td>
      <td>
        row2-col3
      </td>
    </tr>
    <tr
      className="brew-TableRow"
    >
      <td>
        row3-col1
      </td>
      <td>
        row3-col2
      </td>
      <td>
        row3-col3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Renders correctly 1`] = `
<table
  className="brew-Table"
>
  <thead
    className="brew-TableHeader"
  >
    <tr
      className="brew-TableRow"
    >
      <th
        className="brew-TableHeaderColumn"
      >
        Column1
      </th>
      <th
        className="brew-TableHeaderColumn"
      >
        Column2
      </th>
      <th
        className="brew-TableHeaderColumn"
      >
        Column3
      </th>
    </tr>
  </thead>
  <tbody
    className="brew-TableBody"
  >
    <tr
      className="brew-TableRow"
    >
      <td>
        Row1-Col1
      </td>
      <td>
        Row1-Col2
      </td>
      <td>
        Row1-Col3
      </td>
    </tr>
    <tr
      className="brew-TableRow"
    >
      <td>
        Row2-Col1
      </td>
      <td>
        Row2-Col2
      </td>
      <td>
        Row2-Col3
      </td>
    </tr>
  </tbody>
</table>
`;
